Examples/Io/EDM4hep/include/ActsExamples/Io/EDM4hep/EDM4hepMeasurementInputConverter.hpp

```diff
@@ -49,27 +49,20 @@ class EDM4hepMeasurementReader final : public IReader {
   ///
   /// @param config is the configuration object
   /// @param level is the logging level
-  EDM4hepMeasurementReader(const Config& config, Acts::Logging::Level level);
-
-  std::string name() const final;
-
-  /// Return the available events range.
-  std::pair<std::size_t, std::size_t> availableEvents() const final;
+  EDM4hepMeasurementInputConverter(const Config& config,
+                                   Acts::Logging::Level level);
 
   /// Read out data from the input stream.
-  ProcessCode read(const ActsExamples::AlgorithmContext& ctx) final;
+  ProcessCode convert(const AlgorithmContext& ctx,
+                      const podio::Frame& frame) const override;
 
   /// Readonly access to the config
   const Config& config() const { return m_cfg; }
 
  private:
   Config m_cfg;
-  std::pair<std::size_t, std::size_t> m_eventsRange;
-  std::unique_ptr<const Acts::Logger> m_logger;
-
-  tbb::enumerable_thread_specific<Acts::PodioUtil::ROOTReader> m_reader;
 
-  Acts::PodioUtil::ROOTReader& reader();
+  ReadDataHandle<podio::Frame> m_inputFrame{this, "InputFrame"};
```

Comment:
doesn't the base class provide this?

Reply:
This is the input converter: the `PodioReader` will read a `podio::Frame` from a file and store it in the event store. This algorithm reads and converts from the frame that it retrieves from there.